src/fixate/core/checks.py

```diff
@@ -2,72 +2,187 @@
 This module is used to allow for tests to test values against criteria.
 It should implement necessary logging functions and report success or failure.
 """
-import fixate.config
+from dataclasses import dataclass
+from typing import Any, Callable, Iterable
+import logging
 
+import fixate
+
+_logger = logging.getLogger(__name__)
+
+
+@dataclass
```

Comment:
@dataclass(frozen=True) is the behaviour you wanted

Reply:
Yep - done